### PR TITLE
Create README.md

### PR DESCRIPTION
**Pull request by gayashanbc**

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+1. Clone the repository with `git clone https://github.com/gayashanbc/x509-ed25519-grant-handler.git`.
+2. Build using Maven `mvn clean install`.
+
+You can find the `wso2is-identity-samples-oauth2-ssh-ed25519-cert-v01-grant-1.0.0.jar` inside the `target` folder of the project root directory.
+
+### Downloading the release artifact
+1. Download [wso2is-identity-samples-oauth2-ssh-ed25519-cert-v01-grant-1.0.0.jar](https://github.com/gayashanbc/x509-ed25519-grant-handler/releases/download/1.0.0/wso2is-identity-samples-oauth2-ssh-ed25519-cert-v01-grant-1.0.0.jar).
+
+## Configuring
+1. Coyy `wso2is-identity-samples-oauth2-ssh-ed25519-cert-v01-grant-1.0.0.jar` to `<IS_HOME>/repository/components/dropins` directory.
```
Comment (omindu) [inline README.md]:
```suggestion
1. Copy `wso2is-identity-samples-oauth2-ssh-ed25519-cert-v01-grant-1.0.0.jar` to `<IS_HOME>/repository/components/dropins` directory.
```

File: README.md [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+3. Restart WSO2 IS.
+4. Configure a service provider to test the sample under OpenId connect configurations.
+5. You will be able to see "x509" as a grant type under supported grant types and enable it.
+6. Click update on service provider configurations.
+
+## Debugging
+Append the following configurations to `log4j2.properties` file located in `<IS_HOME>/repository/conf`
+
+```
+logger.org-wso2-sample-identity-oauth2-grant-ed25519.name=org.wso2.sample.identity.oauth2.grant.ed25519
```
Comment (omindu) [inline README.md]: This needs to be appended to `loggers` section as well in log4j file

File: README.md [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+grant_handler="org.wso2.sample.identity.oauth2.grant.ed25519.X509GrantHandler"
+grant_validator="org.wso2.sample.identity.oauth2.grant.ed25519.X509GrantValidator"
+[oauth.custom_grant_type.properties]
+IdTokenAllowed=true
+```
+3. Restart WSO2 IS.
+4. Configure a service provider to test the sample under OpenId connect configurations.
+5. You will be able to see "x509" as a grant type under supported grant types and enable it.
+6. Click update on service provider configurations.
+
```
Comment (omindu) [inline README.md:46]: Can we add a sample grant request ?

Comment (gayashanbc) [reply README.md:46]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/4/commits/0884db8a2eaffdf814eab1ec2f63687e735c8db8

File: README.md (hunk 2 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/4/commits/0884db8a2eaffdf814eab1ec2f63687e735c8db8

File: README.md (hunk 1 above)
Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/4/commits/0884db8a2eaffdf814eab1ec2f63687e735c8db8

File: README.md [hunk 4]
```diff
@@ -0,0 +1,53 @@
...
+### Trying out
+Executing the following sample cURL request to try out the grant handler after replacing `<OAuth Client Key>` and `<OAuth Client Secret>` with the respective values.<br/>
+Note that the user principal presented in the sample request certificate: `test_user` needs to exist in a user store for the request to be completed successfully.
+```shell script
+curl -kv \ 
+ --data-urlencode "grant_type=x509" \
+ --data-urlencode "scope=openid" \
+ --data-urlencode "x509=ssh-ed25519-cert-v01@openssh.com AAAAIHNzaC1lZDI1NTE5LWNlcnQtdjAxQG9wZW5zc2guY29tAAAAIL+dMYrVu2I49m5CqY/RkPiDwYppvsGAB0LiNHB/2BD9AAAAIGMKpNu2xYZk1gLzuEPHQTbnlz6RYvvvo3U9ygPOzOC0DeC2s6dj//8AAAABAAAAEHNvbWVfdXNlcl9rZXlfaWQAAAANAAAACXRlc3RfdXNlcgAAAABflxx0AAAAAF+XKtgAAAAAAAAAggAAABVwZXJtaXQtWDExLWZvcndhcmRpbmcAAAAAAAAAF3Blcm1pdC1hZ2VudC1mb3J3YXJkaW5nAAAAAAAAABZwZXJtaXQtcG9ydC1mb3J3YXJkaW5nAAAAAAAAAApwZXJtaXQtcHR5AAAAAAAAAA5wZXJtaXQtdXNlci1yYwAAAAAAAAAAAAAAMwAAAAtzc2gtZWQyNTUxOQAAACBb6BeFLgueUbj6eAwxZ/ajSRTf0NTcbjxfhfmGMluBhwAAAFMAAAALc3NoLWVkMjU1MTkAAABAawGKwVXXAWaFTi9KO4HgZOLq8LcmUuDvb3ScqySBUz+O6euwVWqXu0by8uS66sJnU/1uOzsIs/ZekaaMum/+Dw== user@host" \
```
Comment (omindu) [inline README.md]:
```suggestion
 --data-urlencode "x509=ssh-ed25519-cert-v01@openssh.com AAAA...aErf/+Dw== user@host" \
```

Comment (gayashanbc) [reply README.md]: Resolved in https://github.com/gayashanbc/x509-ed25519-grant-handler/pull/4/commits/8ef1d2cf62516523631ae8d408e8a89893f6720b

File: README.md [hunk 5]
```diff
@@ -0,0 +1,55 @@
...
+4. Configure a service provider to test the sample under OpenId connect configurations.
+5. You will be able to see "x509" as a grant type under supported grant types and enable it.
+6. Click update on service provider configurations.
+7. Once the service provider is saved, you will be redirected to the `Service Provider Details` page. Here, expand the
+    `Inbound Authentication Configuration` section and click the `OAuth/OpenID Connect Configuration` section. Copy the
+    values of  `OAuth Client Key` and `OAuth Client Secret` shown here.
+    ![OAuth Client Credentials](https://user-images.githubusercontent.com/15249242/91567068-27155e00-e962-11ea-8eab-b3bdd790bfd4.png)
+
+### Trying out
+Executing the following sample cURL request to try out the grant handler after replacing `<OAuth Client Key>` and `<OAuth Client Secret>` with the respective values.
```
Comment (omindu) [inline README.md]:
```suggestion
Execute the following sample cURL request to try out the grant handler after replacing `<OAuth Client Key>` and `<OAuth Client Secret>` with the respective values.
```

**Review by omindu — APPROVED** [review]

**Pull request merged**